meshtastic/module_config.proto

```diff
@@ -108,6 +108,52 @@ message ModuleConfig {
     uint32 update_interval = 2;
   }
 
+  /*
+   * Detection Sensor Module Config
+   */
+   message DetectionSensorConfig {
+    /*
+     * Whether the Module is enabled
+     */
+    bool enabled = 1;
+
+    /*
+     * Interval in seconds of how often we can send a message to the mesh when a state change is detected
+     */
+    uint32 minimum_broadcast_secs = 2;
+
+    /*
+     * Interval in seconds of how often we should send a message to the mesh with the current state regardless of changes
+     * When set to 0, only state changes will be broadcasted
+     * Works as a sort of status heartbeat for peace of mind
+     */
+    uint32 state_broadcast_secs = 3;
+
+    /*
+     * GPIO pin to monitor for state changes
+     */
+    uint32 monitor_pin = 4;
+
+    /*
+     * Whether or not the GPIO pin state detection is triggered on HIGH (1)
+     * Otherwise LOW (0)
+     */
+    bool detection_triggered_high = 5;
+
```

Comment:
Do we need to specify if a pullup is present? Some boards may need it, but it may confuse people (again).